test/generate_prim_sig_tests.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/python
```

Comment:
What's the motivation for not just writing unit tests for these?  These tests are all very much boilerplate, but I don't see the need for moving to Python.  One of the reasons I prefer basic tests is that they're easy to find in the code.  With this kind of thing, I'll never be able to find where `math(int, double)` gets tested.  And these should all be tested on a mix of integer and real inputs to make sure promotion and return type calculation works as expected.

And where are we testing that they produce the same values as the C++ functions?

Reply:
The decision to make these tests automatic is mostly because I wanted to test that all scalar argument functions work correctly. 

But I guess we only need to test std:: shadowing functions, which is a non-changing small list of functions. Will revert to that.